### PR TITLE
Issue #43

### PR DESCRIPTION
**Pull request by anuveyatsu**
Implementation of Redux is done here:
* Redux async actions
* Redux reducers
* All Redux setup and configuration
* Tests for Redux actions and reducers

### CODE REVIEW

File: __tests__/src/reducers/datapackageReducer.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import reducer from '../../../src/reducers/datapackageReducer'
+import * as types from '../../../src/constants/actionTypes'
```
Comment (subhankarb) [inline __tests__/src/reducers/datapackageReducer.test.js]: we can remove this line, as we are not using types

File: src/store/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { createStore, applyMiddleware } from 'redux'
+import thunkMiddleware from 'redux-thunk'
+import update from '../reducers/datapackageReducer'
+
+export default function configureStore(preloadedState) {
+  return createStore(
+    update,
```
Comment (subhankarb) [inline src/store/configureStore.js:7]: I think root reducer will be useful. Right now we have only one reducer. If in future we have multiple reducers we can use combine reducer to pass to store. e.g.
```javascript
//reducer/index.js

import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import update from "./datapackageReducer";

const rootReducer = combineReducers({
	routing: routerReducer,
	update: update,
});

export default rootReducer;
```
Then in store/configureStore.js
```javaScript
createStore(rootReducer, initialState, compose(
		applyMiddleware(...middlewares)
		)
```

**Review by subhankarb — COMMENTED** [review]
Small change in reducer. Else LGTM

File: src/store/configureStore.js (hunk 2 above)
Comment (anuveyatsu) [reply src/store/configureStore.js:7]: Yes, I agree with that, but could we leave it for later as it is sort of improvement that is not required right now?

**Pull request merged**